src/core_scan/Core_scan.ml

```diff
@@ -522,7 +522,7 @@ let sanity_check_invalid_patterns (res : Core_result.t) :
 (*****************************************************************************)
 
 (* for -rules *)
-let rules_from_rule_source (config : Core_scan_config.t) :
+let%trace rules_from_rule_source (config : Core_scan_config.t) :
```

Comment:
This is really ugly. Is there a way to use attributes instead of those special let?

Reply:
For example using let%xxx will prevent to use simple semgrep patterns to find definitions
in OCaml; I'd rather use attributes.

Reply:
This is how their ppx works. I don't think there's an alternative. https://github.com/c-cube/ocaml-trace

Reply:
Can you ask them whether they could also support the [@@trace] syntax?

Reply:
Hm, I'm not sure how they'd make it work with their syntax to add data to span, which I do find nice (see `handle_target_with_trace` for an example). I anticipate that we'll use it pretty often to tag with rule or file data that will be useful for filtering; it's the main reason I was determined to switch to their ppx.

Reply:
I agree with @emjin -- this syntax doesn't offend me and I think it's nice to have one way of doing a particular thing. They've chosen this syntax and that is okay with me.